File: Stage6/C/Problem1157.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main() {
    char word[1000001];
    scanf("%s", word);
    int alphabet[26] = {0};
    int len = strlen(word);
    
    for (int i = 0; i < len; i++) {
        alphabet[toupper(word[i]) - 'A']++;
    }

    int maxValue = 0;
    char result = '?';

    for (int i = 0; i < 26; i++) {
        if (alphabet[i] > maxValue) {
            maxValue = alphabet[i];
            result = i + 'A';
        } else if (alphabet[i] == maxValue) {
            result = '?';
        }
    }

    printf("%c\n", result);
    return 0;
}

File: Stage6/Java/Problem1157.java
package Stage6.Java;

import java.util.Scanner;

public class Problem1157 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String word = sc.nextLine().toUpperCase();
        int[] alphabet = new int[26];
        
        for (int i = 0; i < word.length(); i++) {
            alphabet[word.charAt(i) - 'A']++;
        }

        int maxValue = 0;
        char result = '?';
        
        for (int i = 0; i < 26; i++) {
            if (alphabet[i] > maxValue) {
                maxValue = alphabet[i];
                result = (char)(i + 'A');
            } else if (alphabet[i] == maxValue) {
                result = '?';
            }
        }

        System.out.println(result);
        sc.close();
    }
}